broadpeakplugin/src/main/java/com/kaltura/playkit/plugins/broadpeak/BroadpeakPlugin.java

```diff
@@ -146,7 +146,8 @@ public void apply(PKMediaEntry mediaEntry, PKMediaEntryInterceptor.Listener list
         String errorMessage = BroadpeakError.Unknown.errorMessage;
 
         if (mediaEntry != null && mediaEntry.getSources() != null &&
-                !mediaEntry.getSources().isEmpty() && mediaEntry.getSources().get(0) != null) {
+                !mediaEntry.getSources().isEmpty() && mediaEntry.getSources().get(0) != null &&
```

Comment:
Isn't the real issue the fact we have a source without a url? Maybe we need to solve this in the providers where we can validate for sources without url.

Reply:
@OrenMe  after looking again at the exception it is related to session becoming null not the url